keyboards/handwired/xealousbrown/rules.mk

```
@@ -0,0 +1,60 @@
+
+# MCU name
+MCU = atmega32u4
+
+# Processor frequency.
+#     This will define a symbol, F_CPU, in all source code files equal to the
+#     processor frequency in Hz. You can then use this symbol in your source code to
+#     calculate timings. Do NOT tack on a 'UL' at the end, this will be done
+#     automatically to create a 32-bit value in your source code.
+#
+#     This will be an integer division of F_USB below, as it is sourced by
+#     F_USB after it has run through any CPU prescalers. Note that this value
+#     does not *change* the processor frequency - it should merely be updated to
+#     reflect the processor speed set externally so that the code can use accurate
+#     software delays.
+F_CPU = 16000000
+
+
+#
+# LUFA specific
+#
+# Target architecture (see library "Board Types" documentation).
+ARCH = AVR8
+
+# Input clock frequency.
+#     This will define a symbol, F_USB, in all source code files equal to the
+#     input clock frequency (before any prescaling is performed) in Hz. This value may
+#     differ from F_CPU if prescaling is used on the latter, and is required as the
+#     raw input clock is fed directly to the PLL sections of the AVR for high speed
+#     clock generation for the USB and other AVR subsections. Do NOT tack on a 'UL'
+#     at the end, this will be done automatically to create a 32-bit value in your
+#     source code.
+#
+#     If no clock division is performed on the input clock inside the AVR (via the
+#     CPU clock adjust registers or the clock division fuses), this will be equal to F_CPU.
+F_USB = $(F_CPU)
+
+# Interrupt driven control endpoint task(+60)
+OPT_DEFS += -DINTERRUPT_CONTROL_ENDPOINT
+
+
```

Do you know what Boot loader is being used?  

If so, could you specify that rather than the size?  Eg: 
```make
# Bootloader selection
#   Teensy       halfkay
#   Pro Micro    caterina
#   Atmel DFU    atmel-dfu
#   LUFA DFU     lufa-dfu
#   QMK DFU      qmk-dfu
#   atmega32a    bootloadHID
BOOTLOADER = atmel-dfu
```

```suggestion
BOOTLOADER = atmel-dfu
```

---

How do i determine boot loader? I'm using a chinese pro micro. Does wrong bootloader selection brick my board?

---

If it's a Pro Micro, at all, then it's using Caterina (or a variant of it). 

And no, it won't brick the board, if it's wrong. However, it WILL NOT reset into the bootloader properly (when using the RESET keycode, or `reset_keyboard()` function). 

Additionally, this setting is used to determine the bootloader size, and therefore, the max firmware size, so that it properly errors out if it's too large. 

But this should be set to caterina, since it's a Pro Micro. 

```suggestion
BOOTLOADER = caterina
```